src/Avalonia.Controls/TreeView.cs

```diff
@@ -215,8 +215,7 @@ private void SubscribeToSelectedItems()
 
         private void SelectSingleItem(object item)
         {
-            SelectedItems.Clear();
-            SelectedItems.Add(item);
+            SelectedItems = new AvaloniaList<object>(item);
```

Comment:
In best case scenario this will break user code that binds own collection to `SelectedItems` and in worst case it will crash the entire application when we try to bind incompatible types (user has two way binding to `SelectedItems`.)

Reply:
So whats your suggestion then?

Reply:
why not make a copy of existing selection, add the new item, and remove the others?

Reply:
Just perform a replace in case of a single item